tests/Unit/ComponentDependencyInjectionTest.php

```diff
@@ -118,6 +118,51 @@ public function component_render_method_with_dependency()
 
         $component->assertSee('Results from the service');
     }
+
+    /** @test */
+    public function component_mount_action_with_primitive_union_types()
+    {
+        $php_version = phpversion();
+        $php_major_version = explode('.', $php_version)[0];
+        if ($php_major_version < 8) {
+            $this->markTestSkipped("Feature not available in php version < 8.0");
+        }
+
+        eval('
+                class ComponentWithUnionTypes extends Livewire\Component
```

Comment:
If you put this class in a separate file and run this test only on PHP 8.0, this file won't be loaded on lover versions.

Reply:
@jrmajor you're right! thanks!

Reply:
@jrmajor it doesn't seem to work, with `@requires php 8.0` the code is still parsed and throws an `ParseError: syntax error, unexpected '|', expecting variable (T_VARIABLE)` exception

any idea?

Reply:
just found that the problem is with the `@requires php 8.0` part, so I managed to extract the class definition to an external file, but the php version checking is still there

Reply:
Have you tried with proper casing (uppercase “PHP”)?

Reply:
ok, I think it's fixed now